eth/watchers/eventdecoder_test.go

```diff
@@ -0,0 +1,74 @@
+package watchers
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/livepeer/go-livepeer/eth/contracts"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestEventDecoder_FindEventType(t *testing.T) {
+	dec, err := NewEventDecoder(stubBondingManagerAddr, contracts.BondingManagerABI)
+	require.Nil(t, err)
+
+	assert := assert.New(t)
+
+	// Test unknown contract address
+	log := newStubBaseLog()
+	log.Address = common.BytesToAddress([]byte("foo"))
+
+	_, err = dec.FindEventName(log)
+	assert.EqualError(err, "log not from known contract")
+
+	// Check that bondingManagerDecoder is used
+	log = newStubUnbondLog()
+
+	eventName, err := dec.FindEventName(log)
+	assert.Nil(err)
+	assert.Equal("Unbond", eventName)
+}
+
+func TestEventDecoder_Decode(t *testing.T) {
+	dec, err := NewEventDecoder(stubBondingManagerAddr, contracts.BondingManagerABI)
+	require.Nil(t, err)
+
+	assert := assert.New(t)
+
+	// Test unknown contract address
+	log := newStubBaseLog()
+	log.Address = common.BytesToAddress([]byte("foo"))
+
+	var dummyEvent struct {
+		foo string
+	}
+	err = dec.Decode("foo", log, dummyEvent)
+	assert.EqualError(err, "log not from known contract")
+
+	// Check that bondingManagerDecoder is used
+	log = newStubUnbondLog()
+
+	var unbondEvent contracts.BondingManagerUnbond
+	err = dec.Decode("Unbond", log, &unbondEvent)
+	assert.Nil(err)
+}
+
+func TestBondingManagerDecoder_FindEventType(t *testing.T) {
+	dec, err := NewEventDecoder(stubBondingManagerAddr, contracts.BondingManagerABI)
+	require.Nil(t, err)
+
+	assert := assert.New(t)
+
+	// Test unknown event for BondingManager
```

Comment:
Comment needs to be updated, maybe something like "Test unknown event for contract"

Reply:
fixed in 7ce08e7